### PR TITLE
Use CopyDFW for struct field summaries with explicit source.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java [hunk 1]
```diff
@@ -78,6 +79,19 @@ private void makeAttributeCombinerTransformIfAppropriate(SummaryField summaryFie
         }
     }
 
+    /*
+     * This function must be called after makeAttributeCombinerIfAppropriate().
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java:83]:
```suggestion
     * This function must be called after makeAttributeCombinerTransformIfAppropriate().
```

File: config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java [hunk 2]
```diff
@@ -78,6 +79,19 @@ private void makeAttributeCombinerTransformIfAppropriate(SummaryField summaryFie
...
+     */
+    private void makeCopyTransformIfAppropriate(SummaryField summaryField, Schema schema) {
+        if (summaryField.getTransform() == SummaryTransform.NONE) {
+            String source_field_name = summaryField.getSingleSource();
+            ImmutableSDField source = schema.getField(source_field_name);
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java:88]:
```suggestion
            ImmutableSDField source = schema.getField(sourceFieldName);
```

File: config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java (hunk 1 above)
Comment (geirst) [inline config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java:83]:
```suggestion
     * This function must be called after makeAttributeCombinerTransformIfAppropriate().
```

File: config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java [hunk 3]
```diff
@@ -78,6 +79,19 @@ private void makeAttributeCombinerTransformIfAppropriate(SummaryField summaryFie
         }
     }
 
+    /*
+     * This function must be called after makeAttributeCombinerIfAppropriate().
+     */
+    private void makeCopyTransformIfAppropriate(SummaryField summaryField, Schema schema) {
+        if (summaryField.getTransform() == SummaryTransform.NONE) {
+            String source_field_name = summaryField.getSingleSource();
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java:87]:
```suggestion
            String sourceFieldName = summaryField.getSingleSource();
```

File: config-model/src/main/java/com/yahoo/vespa/documentmodel/SummaryField.java [hunk 4]
```diff
@@ -263,6 +263,18 @@ public SummaryField clone() {
         }
     }
 
+    public boolean hasExplicitSingleSource() {
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/vespa/documentmodel/SummaryField.java:266]: Please add a class comment.

**Review by geirst — APPROVED** [review]
👍

File: config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java (hunk 2 above)
Comment (toregge) [reply config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java]: Done.

File: config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java (hunk 1 above)
Comment (toregge) [reply config-model/src/main/java/com/yahoo/schema/processing/SummaryConsistency.java]: Done.

File: config-model/src/main/java/com/yahoo/vespa/documentmodel/SummaryField.java (hunk 4 above)
Comment (toregge) [reply config-model/src/main/java/com/yahoo/vespa/documentmodel/SummaryField.java:270]: Comment added.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**